### ISSUE TITLE
Add custom icon and/or color for same category tabs

### ISSUE THREAD

**Issue by kimbap77** [RealRaven2000/QuickFolders#222]
It would be nice if I could assign a customized icon and/or color to a tab category. For example, all the tabs assigned to my 'Projects' category could have the same icon and/or color and my 'Coworkers' category could have a different icon and/or color.
I realize that there would be an issue when assigned to multiple categories. Perhaps, the categories could be ranked? Thanks

**Comment by RealRaven2000** [comment]
Yes I can definitely imagine this being useful. Since assignment of multiple icons is difficult, there needs to be indeed a single icon / color picked for a tab (without overcomplicating things). Then we will need the following features:

- icon picker in the category dialog + checkbox to enable it
- color picker in category dialog + checkbox to enable it
- priority number text field (should this also be used for the order of items within the category picker, or should this stay alphabetical (I need to check whether it is even ordered right now or whether it is populated in the order that new categories "come in" while reading the tabs in their assigned order.

Suggested Behavior: 
- if a tab is assigned no _individual_ color, it will be colored according to the (most important) category
- if a tab has not assigned an _individual_ icon, an icon will be added according to the (most important) category
- if a tab has assigned multiple categories, the most important category will determine color / icons

Additional thought: as long as you do not select all, all your tabs will automatically be colored with the currently selected category / categories (except for the individually colored tabs) - if we ignore selecting all ever, this may actually reduce the inital complexity greatly. Personally I never use "all" anymore - I have simply too many tabs declared. Any thoughts on implementing it in a "simple way" first?
